artmanflow/steps/common.py

```diff
@@ -0,0 +1,367 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import errno
+import tempfile
+import uuid
+import os
+import subprocess
+import sys
+from ruamel import yaml
+
```

Comment:
High level note: Please document all classes and methods with useful and descriptive docstrings. (I know artman has been less than great about that, but it is important, especially for large PRs like this.)

Reply:
It will be documented once/if this becomes more public. Here I'm trying to address comments and then move this to a separate repository.